### PR TITLE
[PublishBuildArtifactsV1] Migrate to Node 10

### PR DESCRIPTION
**Pull request by DaniilShmelev**
**Task name**: PublishBuildArtifacts

**Description**:
- Updated dependencies; removed unused `@types/q` dependency
- Changed typescript version to 4.0.2
- Changed deprecated MochaDone to Mocha.Done in L0.ts
- Removed definitions reference from L0.ts
- Bumped task version

**Documentation changes required:** No

**Added unit tests:** No

**Attached related issue:** https://github.com/microsoft/build-task-team/issues/708 (internal)

**Checklist**:
- [X] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [X] Checked that applied changes work as expected

[Successful pipeline](https://dev.azure.com/v-dshmelev/playground/_build/results?buildId=659&view=results)
[Node 6 hotfixes branch](https://github.com/microsoft/azure-pipelines-tasks/tree/node6hotfixes/PublishBuildArtifactsV1)

### CODE REVIEW

**Review by EzzhevNikita — COMMENTED** [review]
I think it make sense to add some kind of real L0 test scenarios for this task. May be just migrate existing [Legacy suite of tests](https://github.com/microsoft/azure-pipelines-tasks/tree/master/Tests-Legacy/L0/PublishBuildArtifacts).

File: Tasks/PublishBuildArtifactsV1/task.json [hunk 1]
```diff
@@ -12,8 +12,8 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 1,
-        "Minor": 158,
-        "Patch": 3
+        "Minor": 182,
```
Comment (EzzhevNikita) [inline Tasks/PublishBuildArtifactsV1/task.json:15]: Could you please bump task version according current sprint?

**Review by anatolybolshakov — APPROVED** [review]

**Review by EzzhevNikita — APPROVED** [review]

**Review by egor-bryzgalov — CHANGES_REQUESTED** [review]
I've tried to run the following definition to test the task:
```
- bash: |
    echo 'test' > test.txt

- task: PublishBuildArtifactsm@1
  displayName: 'Ordinal publish to a container'
  inputs:
    PathtoPublish: 'test.txt'
    ArtifactName: 'drop'
    publishLocation: 'Container'

- bash: |
    mkdir TestDir

- task: PublishBuildArtifactsm@1
  displayName: 'Copy to local path'
  inputs:
    PathtoPublish: 'test.txt'
    ArtifactName: 'art'
    publishLocation: 'FilePath'
    TargetPath: './TestDir'
```
And it got stuck in a loop. It looks like a bug. @DaniilShmelev Could you please take a look?
[Here's the failed build.](https://dev.azure.com/v-egbryz/TestProject/_build/results?buildId=1563&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=ed22b878-bfa9-5263-daa1-ad412f4e75e4)

**Comment by DaniilShmelev** [comment]
> I've tried to run the following definition to test the task:
> 
> ```
> - bash: |
>     echo 'test' > test.txt
> 
> - task: PublishBuildArtifactsm@1
>   displayName: 'Ordinal publish to a container'
>   inputs:
>     PathtoPublish: 'test.txt'
>     ArtifactName: 'drop'
>     publishLocation: 'Container'
> 
> - bash: |
>     mkdir TestDir
> 
> - task: PublishBuildArtifactsm@1
>   displayName: 'Copy to local path'
>   inputs:
>     PathtoPublish: 'test.txt'
>     ArtifactName: 'art'
>     publishLocation: 'FilePath'
>     TargetPath: './TestDir'
> ```
> 
> And it got stuck in a loop. It looks like a bug. @DaniilShmelev Could you please take a look?
> [Here's the failed build.](https://dev.azure.com/v-egbryz/TestProject/_build/results?buildId=1563&view=logs&j=12f1170f-54f2-53f3-20dd-22fc7dff55f9&t=ed22b878-bfa9-5263-daa1-ad412f4e75e4)

This issue also happens with the old version of the task. I've created a separate [ticket](https://github.com/microsoft/build-task-team/issues/791) (internal).

**Review by egor-bryzgalov — APPROVED** [review]

**Pull request merged**